app/src/org/commcare/android/view/EntityView.java

```diff
@@ -530,5 +491,8 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
                 views[i].setLayoutParams(params);
             }
         }
-    }*/
+        
+        // Use children's new widths
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec);

```

Comment:
@ctsims you're right, I just tried it and it failed horribly!
I added some in-code comments describing what you and @orangejenny elaborated here.